docs/developer-guide/custom-layers/prop-types.md

```diff
@@ -161,6 +162,25 @@ MyLayerClass.defaultProps = {
 }
 ```
 
+##### `object` {#array}
+
+An arbitrary object.
+
+- Options:
+  + `optional` (boolean, optional) - accept `null` or `undefined`. Default `false`.
+  + `ignore` (boolean, optional) - if `true`, prop comparison is disabled. Default `false`.
```

Comment:
Some prop changes are already ignored. For example, changing `onHover` does not trigger a layer update, it will only be applied next time an hover event occurs.

Similarly, layers never handled changes in `loaders` and `loadOptions`. They are only used when data URL changes.

Reply:
> it's impossible to change this prop after it has been set once.

They can be changed, just that the change alone does not trigger any layer action (update or redraw) because they are "passive". Their values will be respected when other situations arise that do require a layer action.